src/client/components/annotations.ts

```diff
@@ -69,27 +94,68 @@ export class Annotations extends LitElement {
 
   // Declare reactive properties
   @property({ type: Object, reflect: true })
-  annotations?: NotebookCellAnnotations
+  annotations?: CellAnnotations
+
+  @property({ type: Object, reflect: true })
+  validationErrors?: CellAnnotationsErrorResult
+
+  @state()
+  private shouldRender: boolean = false
 
   #desc(id: string): string {
     return this.#descriptions.get(id) || id
   }
 
+  #getTargetValue(e: Target) {
+    switch (e.target.type) {
+      case 'text':
+        return e.target.value
+      default:
+        return e.target.checked.toString()
+    }
+  }
+
+
   #onChange(e: { target: { id: AnnotationsKey, checked: boolean, value: string, type: string } }) {
     if (!this.annotations || !e.target) {
       return
     }
 
     const propVal: any = { 'runme.dev/uuid': this.annotations['runme.dev/uuid'] }
     const propName = e.target.id
+    const value = this.#getTargetValue(e)
+    const targetValue = tryBoolean(value)
+
+    const parseResult = CellAnnotationsSchema.safeParse({
+      [propName]: targetValue
+    })
+
+    if (!parseResult.success) {
+      const { fieldErrors } = parseResult.error.flatten()
+      if (this.validationErrors && !this.validationErrors?.errors) {
+        this.validationErrors.errors = {}
+      }
+      if (this.validationErrors?.errors && !this.validationErrors.errors[propName]) {
+        this.validationErrors.errors[propName] = fieldErrors[propName]
+      }
+      // Re-render the form
+      return this.shouldRender = !this.shouldRender
+    }
+
+    if (this.validationErrors?.errors && this.validationErrors.errors[propName]) {
+      delete this.validationErrors.errors[propName]
+      // Re-render the form
+      this.shouldRender = !this.shouldRender
```

Comment:
I don't like how I'm mutating this object here, suggestions are welcome.

Reply:
for re-render please try `this.requestUpdate()`